assets/src/editor/events/hocs/with-editor-event-venue-entity.js

```diff
@@ -0,0 +1,25 @@
+import { withSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+const DEFAULT_OBJECT = {
+	venueEntity: null,
+};
+
+export default createHigherOrderComponent(
+	withSelect( ( select, { eventEntity } ) => {
+		const { getRelatedEntities } = select( 'eventespresso/core' );
+		if ( isModelEntityOfModel( eventEntity, 'event' ) ) {
+			const venueEntity = getRelatedEntities(
+				eventEntity,
+				'venue'
+			);
+			return Array.isArray( venueEntity ) && venueEntity[ 0 ] &&
+				isModelEntityOfModel( venueEntity[ 0 ], 'venue' ) ?
+				{ venueEntity: venueEntity[ 0 ] } :
+				DEFAULT_OBJECT;
+		}
+		return DEFAULT_OBJECT;
+	} ),
+	'withEditorEventVenueEntity'
+);
```

Comment:
This creates a tight coupling of the hoc to getting a venue related to the given event.  Since currently, there's only ever one venue for an event, I think you should move this to be defined as a part of the initialization layer so the venue is retrieved at the start of the editor initialization and then introduce a hoc here that retrieves the current venue in the state.

I _do_ realize that eventually we will make venues related to date-times and that will change things, but without there being any timeline on that eventuality I think we should code things for how they are _now_.  Right now, this means that any component displaying the venue (in this case date entity components) will need to also include `withEditorEventEntity`  in their hoc chain.